### DISCUSSION TITLE
hast-util-raw incorrect import

### DISCUSSION THREAD

**Discussion by DaniFoldi** [syntax-tree/unist#79] [Q&A] [answered]
Hi :wave:,

I can't fully explain _why_ this just broke, it was working fine for a long time, however digging into `hast-util-raw` I found that it has this line:

```js
// @ts-expect-error: untyped.
import Parser from 'parse5/lib/parser/index.js'
```

Changing `lib` to `dist` allowed rollup to resolve the import correctly, as is specified in the `exports` field of `parse5`'s `package.json`.

Is there anything that could be causing import resolution to fail that I missed, or was this somehow missed and should be fixed?

**Comment by wooorm** [comment, marked as answer]
Hey!
I think this is because `parse5` in `hast-util-raw` is on 6. You are somehow forcing 7 in there.
That shouldn’t occur. Depending on what you did and which package manager you are using, there should be a solution.

I tried updating to v7 a month ago, but a lot had changed in the internals, which `raw` deeply depends on. At some point I’ll try again but not likely very soon

**Reply by DaniFoldi** [reply]
Thanks for your quick answer 🙏 !

Appears that indeed `parse5` had version 7.0.0 buried in `node_modules`. I've got a `patch-package` patch that at least got things to _build_, I haven't verified anything yet. In some cases, instead of importing, I've just copied constants from util files.

In this state it's completely possible that some refactoring wasn't taken into account and everything will break majorly, however I'll leave the patch here for future reference in case it turns out to be helpful.

[hast-util-raw+7.2.2.patch](https://github.com/syntax-tree/unist/files/9365244/hast-util-raw%2B7.2.2.patch.txt)
